nixos/modules/services/misc/synergy.nix

```diff
@@ -70,6 +70,26 @@ in
           type = types.bool;
           description = "Whether the Synergy server should be started automatically.";
         };
+        tls = {
+          enable = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              Whether TLS encryption should be used.
+
+              Using this requires a TLS certificate that can be
+              generated by starting the Synergy GUI once and entering
+              a valid product key.
+            '';
+          };
+
+          cert = mkOption {
+            type = types.nullOr types.path;
+            default = null;
+            example = ~/.synergy/SSL/Synergy.pem;
```

Comment:
~ is relative to the user run. I think an absolute path would be better.

Reply:
Thanks for the feedback. I used this path because it is the default used by Synergy if you don't pass an option...

But that made me realize that this would result in inconsistent behavior. If you don't specify anything, a cert from a mutable location is used. If you _do_ specify a path (could even be the same one), that cert will get copied to the store and used from there.

So, I'm guessing it would be better to change this data type to a string. What do you think?

And if I change it, should I still change the example path?

Reply:
Does `home-manager` have a `synergy` module? Keep in mind this module produces `systemd` **user** services... so whatever value you specify here applies to all users on a system. You might run a single user system, but you shouldn't pretend everyone else does. So what makes more sense? I'm not sure. This type of stuff usually goes into `home-manager` instead.

You could make an argument both ways, but I think I might agree with @SuperSandro2000 that an absolute path might be a better general example. It is just an example though... so don't stress out too much about it, it doesn't matter too much either way :smile: 

`types.string` sounds good :+1:

Reply:
Those are some good points.

Taking all the feedback into consideration, I think it's probably best to turn this into a string and leave the example path as is, since that is the actual default when the option is not provided. This way it's possible to provide a global cert for all users by specifying an absolute path, if desired, or alternatively use per-user certs relative to the home directory (which makes most sense when I think about it).

Btw, I didn't find a home-manager module for synergy.